### PR TITLE
jwa(front): Create distinct notebook details page

### PR DESCRIPTION
**Pull request by orfeas-k**
This PR is part of the effort to add [details pages for objects in our WAs](https://github.com/kubeflow/kubeflow/issues/6707). It creates **a new notebook details page where a user can navigate and see key information about each notebook**. More specifically, this tab contains an overview of the `.status` of the object plus a selection of the most important fields from the spec. These are _Volumes, Applied Configurations from PodDefaults, Notebook type, ENV Vars, Minimum/Maximum CPU/RAM, Docker image, Conditions_. We will also introduce soon `LOGS`, `EVENTS` and `YAML` tab.

### Route
Regarding **the link** that this page will be served on, we need to be able to extract from the link information about the **notebook's name** we 're currently viewing and the **current namespace**.  Thus, we ended up with `notebook/details/namespace/notebook-name?ns=namespace` .

### Notebook's Conditions
Our plan here was to add a Conditions table inside the `OVERVIEW` tab in order to show the `.status.conditions` of the notebook. 

**Problem**
However, we found out that the notebook’s `.status.conditions` are fundamentally broken. Right now the Notebook Controller is blindly appending Pod conditions on the Notebook's conditions https://github.com/kubeflow/kubeflow/blob/master/components/notebook-controller/controllers/notebook_controller.go#L247. 
The conditions are expected to provide a summary of the state of an object for a specific point in time, not use the list to combine the conditions to get to the final state. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

**Workaround**
Since currently, there is no point in exposing the notebook `.status.conditions`, we decided, as a temporary workaround, to show the Notebook’s underlying Pod  conditions and to be explicit about which conditions the table refers to.

### Sidenote
For **clickable items - links**, we followed the style and what we do in general in KFP (e.g. Output Artifacts)

### Screenshots 
image
image
image

cc @kimwnasptd

### CODE REVIEW

File: components/crud-web-apps/jupyter/frontend/src/app/pages/notebook-page/overview/overview.component.html [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  <lib-panel class="page-padding">
+    {{ podConditionsMessage }}
+  </lib-panel>
+</ng-container>
+
+<!--Show Pod Conditions-->
+<ng-container *ngIf="podRequestCompleted && pod">
+  <lib-conditions-table
+    *ngIf="pod?.status"
+    [conditions]="pod.status?.conditions"
```
Comment (kimwnasptd) [inline components/crud-web-apps/jupyter/frontend/src/app/pages/notebook-page/overview/overview.component.html:70]: Let's use the conditions from the Notebook status.

We've updated the Notebook Controller to properly set the conditions in https://github.com/kubeflow/kubeflow/pull/6628

**Review by kimwnasptd — COMMENTED** [review]
Looks great @orfeas-k!

Had some proposals mostly around using Notebook's Conditions and not the Pod ones. We've ensured the Controller is setting the Notebook Conditions properly from the previous release cc @apo-ger

**Comment by orfeas-k** [comment]
@kimwnasptd Had to force push since I rebased to resolve the conflicts but the fixes for the comments above are separate commits on their own.

**Comment by kimwnasptd** [comment]
Changes look good, thanks for driving this @orfeas-k!

/lgtm
/approve

**Pull request merged**